rust/main/agents/scraper/src/store/storage.rs

```diff
@@ -249,14 +255,10 @@ impl HyperlaneDbStore {
                 continue;
             }
 
-            self.db
-                .store_blocks(
-                    self.domain.id(),
-                    blocks_to_insert
-                        .iter_mut()
-                        .map(|(_, info)| info.take().unwrap()),
-                )
-                .await?;
+            let blocks = blocks_to_insert
+                .iter_mut()
+                .filter_map(|(_, info)| info.take());
```

Comment:
Possibly, it is easier to fix it by changing type of `blocks_to_insert` from `Vec<(&mut BasicBlock, Option<BlockInfo>)>`  to `Vec<(&mut BasicBlock, BlockInfo)>`

Reply:
Updated to 2 vecs. I think its much more ergonomic this way.
The previous way tried to create an iterator from the vec, then drain the vec.
So makes more sense to just separate them into 2 vecs